shared/obs-shared-memory-queue/shared-memory-queue.c

```diff
@@ -0,0 +1,225 @@
+#include <windows.h>
+#include "shared-memory-queue.h"
+#include "tiny-nv12-scale.h"
+
+#define VIDEO_NAME L"OBSVirtualCamVideo"
```

Comment:
Conceptually, this feels a bit weird because `shared-memory-queue` was obviously written to only be used by the Virtual Camera on Windows, but now it's in `shared` and not in a "Windows-only" project/folder. Though, having said that, it looks like win-dshow didn't early return when not targeting Windows either, so maybe that concern is invalid.

Reply:
Yeah the main issue that `shared-memory-queue` is a set of utility functions used by `win-dshow` to communicate with the virtual camera. It is shared by both targets, but the targets have no direct dependency on each other.

So once again that code fits the very definition of "shared" code. See also my comment regarding the "utility library" which would be the correct fix, but lots of Windows code had been written with this implicit sharing of source code and headers across target/project boundaries.